### DISCUSSION TITLE
measureElement default doesn't include margins

### DISCUSSION THREAD

**Discussion by kand** [TanStack/virtual#734] [Ideas]
We have a virtual list that can get new, dynamically sized items in both directions. We were having a lot of scrolling issues until I realized that the default implementation of [`measureElement`](https://tanstack.com/virtual/latest/docs/api/virtualizer#measureelement-1), which uses [`getBoundingClientRect()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect), does not include margins. Switching these margins to padding fixed most of the scrolling issues. Given the gap between items could be accomplished with margins _or_ padding, and the choice is not immediately obvious, is this something that could be highlighted in the documentation? Or could the default implementation include margins?

**Comment by piecyk** [comment]
>  Given the gap between items could be accomplished with margins or padding, and the choice is not immediately obvious, is this something that could be highlighted in the documentation? Or could the default implementation include margins?

That is true, `getBoundingClientRect()` does not include margins, which can cause layout problems with dynamically sized items. Highlighting this in the documentation is a great idea, and we'd be happy to accept a contribution for this clarification. Also, the `gap` option can be used for default spacing to avoid these issues.
